src/utils.c

```diff
@@ -339,6 +340,109 @@ timestamptz_bucket(PG_FUNCTION_ARGS)
 	PG_RETURN_TIMESTAMPTZ(result);
 }
 
+TS_FUNCTION_INFO_V1(ts_int16_bucket);
```

Comment:
Can we move this stuff out of `utils.c`? Either a `time_bucket.c` file (seems warranted) or some analytics-related file.

I don't really know why the other functions, like `date_bucket`, were put here to begin with.